app/models/host_info_providers/static_info.rb

```diff
@@ -8,12 +8,12 @@ def host_info
       param["fqdn"] = host.fqdn unless host.fqdn.nil?
       param["hostgroup"] = host.hostgroup.to_label unless host.hostgroup.nil?
       param["comment"] = host.comment if host.comment.present?
+      param["root_pw"] = host.root_pass unless (!host.operatingsystem.nil? && host.operatingsystem.password_hash == 'Base64')
```

Comment:
I know you're just moving it, but I found this horribly complex to read. Trying to simplify it (removing some negations)
```suggestion
      param["root_pw"] = host.root_pass if (host.operatingsystem.nil? || host.operatingsystem.password_hash != 'Base64')
```

And is that then the same as:

```suggestion
      param["root_pw"] = host.root_pass if host.operatingsystem&.password_hash != 'Base64'
```

Reply:
I agree the current condition is confusing... However, the current implementation actually includes the `root_pw` if the os is `nil`. Not sure if that is correct behavior or not, but I'd rather not modify it in this PR.